jujugui/static/gui/src/app/components/deployment-flow/credential/credential.js

```diff
@@ -77,30 +75,57 @@ YUI.add('deployment-credential', function() {
       @method _getCredentials
     */
     _getCredentials: function() {
-      this.credentialXHR = this.props.listTemplates(
-        this._getCredentialsCallback);
+      const cloud = this.props.cloud;
+      const user = this.props.users.jem.user;
+      if (user) {
+        this.props.getTagsForCloudCredentials(
+          [[user, `cloud-${cloud}`]],
+          this._getTagsCallback);
+      }
     },
 
     /**
-      The method to be called when the credentials reponse has been received.
+      The method to be called when the credentials tags reponse has been
+      received.
+
+      @method _getTagsCallback
+      @param {String} error An error message, or null if there's no error.
+      @param {Array} tags A list of the tags found.
+    */
+    _getTagsCallback: function(error, tags) {
+      if (error) {
+        console.error('Unable to get tags for credentials', error);
+        return;
+      }
+      const tagList = [];
+      tags.forEach(tag => {
+        tagList = tagList.concat(tag.tags);
```

Comment:
Probably a dumb question, but don't we need credentials based on the selected user/cloud pair? ISTM that you are flattening them here.